demo/mobile/src/app/app.js

```diff
@@ -22,6 +25,112 @@ angular.module('wfm-mobile', [
   }),
   require('@raincatcher/vehicle-inspection')
 ]);
+
+var auth = {};
+
+// initialise the keycloak js adapter if keycloak configuration has been specified
+if (keycloakConfig) {
+  // require keycloak js client adapter
+  var Keycloak = require('keycloak-js');
+
+  // keycloak init config
+  var initConfig = {onLoad: 'login-required'};
+
+  // the keycloak json config
+  var keycloakJS = Keycloak(keycloakConfig);
+
+  angular.element(document).ready(function() {
+    // initialise the Keycloak JS Adapter
+    keycloakJS.init(initConfig).success(function() {
+      auth.keycloak = keycloakJS;
+
+      console.log("Keycloak Ininitalisation Success");
+
+      // load keycloak user profile
+      auth.keycloak.loadUserProfile().success(function(profile) {
+        auth.keycloak.userProfile = profile;
+      }).error(function(err) {
+        console.log("Failed to Load User Profile", err);
+      });
+
+      auth.keycloak.getUserId = function() {
+        var userId = auth.keycloak.userProfile.attributes.id[0];
+        return Promise.resolve({"id": userId});
```

Comment:
In old solution we no longer use userId in the demo.
We just pass session token. Having this method makes no sense now - I left comment in previous PR about that, but I believe that we should have conversation about how we going to do this properly and what kind of level of security we want to provide for sync based api. 

💯 Really important topic!